emulsify.drush.inc

```diff
@@ -0,0 +1,299 @@
+  $machine_name = drush_get_option('machine_name');
+  if (!$machine_name) {
+    $machine_name = $name;
+  }
+  $machine_name = str_replace(' ', '_', strtolower($machine_name));
+  $search = array(
+    // Remove characters not valid in function names.
+    '/[^a-z0-9_]/',
+    // Functions must begin with an alpha character.
+    '/^[^a-z]+/',
+  );
+  $machine_name = preg_replace($search, '', $machine_name);
+
+  // Description of theme.
+  $description = (drush_get_option('description')) ? trim(drush_get_option('description')) : 'Theme that uses Pattern Lab v2.';
+
+  // Determine the path to the new theme.
+  $theme_path = 'themes/custom';
+  if ($path = drush_get_option('path')) {
+    $theme_path = drush_trim_path($path);
+  }
+
+  drush_emulsify_create($name, $machine_name, $description, $theme_path);
+}
+
+/**
+ * Create frontend theme.
+ */
+function drush_emulsify_create($name, $machine_name, $description, $theme_path) {
+  $theme_path = drush_normalize_path(drush_get_context('DRUSH_DRUPAL_ROOT') . '/' . $theme_path . '/' . $machine_name);
+
+  if (!is_dir(dirname($theme_path))) {
+    drush_die(dt('The directory "!directory" was not found.', array('!directory' => dirname($theme_path))));
+  }
+
+  // Phase 1: Copy files.
+  $files_to_copy = _emulsify_get_files_to_copy();
+  _emulsify_copy_files($files_to_copy, $theme_path);
+
+  // Phase 2: Alter files.
+  $alterations = _emulsify_get_alterations($name, $machine_name);
+  $files_to_alter = _emulsify_get_files_to_alter();
+  _emulsify_alter_files($theme_path, $files_to_alter, $alterations);
+
+  // Phase 3: Rename files.
+  $files_to_rename = _emulsify_get_files_to_rename();
+  _emulsify_rename_files($theme_path, $machine_name, $files_to_rename);
+
+  _emulsify_notify($name, $theme_path);
+}
+
+/**
+ * Gets alterations (string replacements).
+ *
+ * This function supports both directories and individual files.  Alterations
+ * happen in sequential order so you can replace something that was previously
+ * replaced.
+ */
+function _emulsify_get_alterations($name, $machine_name) {
+  return array(
+    'Emulsify' => $name,
+    'emulsify' => $machine_name,
+    'hidden: true' => '',
+  );
+}
+
+/**
+ * Returns an array of files to make string replacements.
+ */
+function _emulsify_get_files_to_alter() {
+  // Slim files and directories declaration.
+  $slim_array = array(
+    'components',
+    'emulsify.info.yml',
+    'emulsify.theme',
+  );
+  // If we would like to have a bare copy we use is slim option.
+  if (drush_get_option('slim') === TRUE) {
+    return $slim_array;
+  }
+  else {
+    return array_merge($slim_array, array(
+      'templates',
+      'emulsify.breakpoints.yml',
+      'emulsify.libraries.yml',
+      'README.md',
+    ));
+  }
+}
+
+/**
+ * Gets files to copy.
+ *
+ * This function supports both directories and individual files.
+ *
+ * The following directories/files will never be copied:
+ * css/
+ * dist/
+ * node_modules/
+ * github.com/
+ * composer.json
```

Comment:
@ModulesUnraveled I took your advice on the other comments.  I am actually not too sure about this one.  Should we leave it open until you have a chance to work on the composer stuff?